### ISSUE TITLE
Weird behaviour with porcelain ls_remote

### ISSUE THREAD

**Issue by tkw1536** [jelmer/dulwich#471]
I am attempting to use ls_remote as follows:
```python
from dulwich import porcelain
porcelain.ls_remote("https://github.com/jelmer/dulwich.git")
```
However, when I run this I get the following exception: 
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/user/tmp/dulwich/venv/lib/python3.5/site-packages/dulwich-0.16.2-py3.5-linux-x86_64.egg/dulwich/porcelain.py", line 907, in ls_remote
    return client.get_refs(encode_path(host_path))
  File "/home/user/tmp/dulwich/venv/lib/python3.5/site-packages/dulwich-0.16.2-py3.5-linux-x86_64.egg/dulwich/client.py", line 1219, in get_refs
    url = self._get_url(path)
  File "/home/user/tmp/dulwich/venv/lib/python3.5/site-packages/dulwich-0.16.2-py3.5-linux-x86_64.egg/dulwich/client.py", line 1068, in _get_url
    return urlparse.urljoin(self._base_url, path).rstrip("/") + "/"
  File "/usr/lib64/python3.5/urllib/parse.py", line 415, in urljoin
    base, url, _coerce_result = _coerce_args(base, url)
  File "/usr/lib64/python3.5/urllib/parse.py", line 111, in _coerce_args
    raise TypeError("Cannot mix str and non-str arguments")
TypeError: Cannot mix str and non-str arguments
```

I tried debugging this myself and surprisingly the following code worked without problems:
```python
from dulwich import client
gclient, host_path = client.get_transport_and_path("https://github.com/jelmer/dulwich.git")
gclient.get_refs(host_path)
```

I am using Python 3.5.2 on a 64 bit linux machine with a virtualenv at ```/home/user/tmp/dulwich/venv```.  I installed the latest version of dulwich from master (which at the time of writing is f2fcf99a03533769208dd362c8306b4e5dd580c6 ).

**Comment by jelmer** [comment]
This should be fixed in master.

**Comment by dzhuang** [comment]
This was fixed in 6676b7f6c66a1ec4e352ac3256a62e00b551c165
